### ISSUE TITLE
Docker builds fail while trying to install SmartDeployer

### ISSUE THREAD

**Issue by rudi** [abstools/abstools#322]
**Update**: Docker has cancelled its plans; renaming this bug to track the collaboratory image build failure.

*Original bug text:*

Docker will remove our images within the month; move docker images to the github container registry.

Documentation at https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry

- [ ] Rename containers, test building them
- [ ] Update RELEASE.md
- [ ] Update website
- [ ] Upload last released version of absc, collaboratory to ghcr.io
- [ ] (optional) set up github workflow that automatically updates container images

**Comment by rudi** [comment]
It seems the older docker images do not build anymore, due to a failure in `pip install toposort`; this seems to be part of SmartDeployer.

@jacopoMauro do you happen to know if this is still  supposed to work / build?  You can try to build either the current image (`docker build . -t abslang/collaboratory -f docker/collaboratory.Dockerfile`) or check out the last released version (`git checkout v1.9.3 ; docker build ...`); neither of them currently builds.

The tail of the output of the failing build is:

```
 > [stage-3 15/18] RUN pip install         antlr4-python2-runtime         toposort         psutil         click:
#32 0.946 Collecting antlr4-python2-runtime
#32 1.101   Downloading https://files.pythonhosted.org/packages/7b/0f/f604e36b51fe2219efdc009fc9676081a7f6be8f4bc21fd3486c893739d3/antlr4-python2-runtime-4.12.0.tar.gz (113kB)
#32 1.330 Collecting toposort
#32 1.367   Downloading https://files.pythonhosted.org/packages/69/19/8e955d90985ecbd3b9adb2a759753a6840da2dff3c569d412b2c9217678b/toposort-1.10.tar.gz
#32 1.376   Installing build dependencies: started
#32 3.579   Installing build dependencies: finished with status 'done'
#32 3.776     Complete output from command python setup.py egg_info:
#32 3.776     Traceback (most recent call last):
#32 3.776       File "<string>", line 1, in <module>
#32 3.776     IOError: [Errno 2] No such file or directory: '/tmp/pip-install-C6yssQ/toposort/setup.py'
#32 3.776     
#32 3.776     ----------------------------------------
#32 3.796 Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-C6yssQ/toposort/
------
executor failed running [/bin/sh -c pip install         antlr4-python2-runtime         toposort         psutil         click]: exit code: 1
```

**Comment by rudi** [comment]
We will stay with the docker registry until further notice, but switching to github's container registry is quite straightforward and can be done on relatively short notice.
